src/Shared/ReuseableStringBuilder.cs

```diff
@@ -88,6 +97,10 @@ bool OpportunisticIntern.IInternable.IsOrdinalEqualToStringOfSameLength(string o
 #if DEBUG
             ErrorUtilities.VerifyThrow(other.Length == _borrowedBuilder.Length, "should be same length");
 #endif
+            if (other.Length > 40_000)
```

Comment:
Can you please
* Explain in a comment how this number was derived
* Remove the `_` (it's better for readability but evidently failing our Mono builds, probably because of a slightly stale compiler there)